### DISCUSSION TITLE
Question about the shape of Steinhardt's parameters

### DISCUSSION THREAD

**Discussion by ken-wooo** [pyscal/pyscal#152] [Q&A]
Dear,

Many thanks for your very nice package, which helps me a lot!

When I followed the example 'Calculating bond orientational order parameters' via pyscal3, I found that the bcc structures created with 128 atoms have 1024 Steinhardt's parameters, while the fcc structures created with 256 atoms have only 256 Steinhardt's parameters.

So what is the meaning of the shape of the calculated Steinhardt's parameters and why do sometimes the Steinhardt's parameters have 8*natoms size?

I really appreciate any help you can provide.

**Comment by srmnitc** [comment]
@ken-wooo Thanks for starting this discussion. The results should be n_s*n_atoms; where n_s is the number of Steinhardt parameters. For the following example from the docs:

```
bcc = System.create.lattice.bcc(lattice_constant=3.147, repetitions=[4,4,4])
fcc = System.create.lattice.fcc(lattice_constant=3.147, repetitions=[4,4,4])

bcc.find.neighbors(method='cutoff', cutoff=3.6)
fcc.find.neighbors(method='cutoff', cutoff=2.7)

bccq = bcc.calculate.steinhardt_parameter([4,6])
fccq = fcc.calculate.steinhardt_parameter([4,6])
```

The number of Steinhardt paramters should be `2*256` for fcc, and `2*128` for bcc. The reason that you get more for bcc is because pyscal uses an internal representation with more atoms since the bcc cell is small. When you access the calculated parameter through:

```
bcc.atoms.steinhardt.generic.q4_norm
```
you get the right shape as the internal atoms are filtered out. 

`bccq` from the above example should also have the same shape. Thanks for pointing this out. I will start an issue and fix this.

**Reply by srmnitc** [reply]
Started issue over at #153
